### PR TITLE
update faster plugin

### PR DESCRIPTION
**Pull request by bettybao1209**
Related to PR #158 
1. upgrade fasterdb version to `2020.1.14.1`,  and update corresponding codes;
2. add serialnum(sno) to distinguish write operations between store and snapshot.

### CODE REVIEW

File: src/FasterDBStore/Store.cs [hunk 1]
```diff
@@ -142,22 +143,33 @@ public IEnumerable<(byte[] Key, byte[] Value)> Find(byte table, byte[] prefix)
...
+
             var k = new BufferKey(table, key);
             var v = new BufferValue(value);
 
-            if (db.Upsert(ref k, ref v, Empty.Default, 0) == Status.PENDING)
+            session.Upsert(ref k, ref v, Empty.Default, 0);
+            var status = session.Read(ref k, ref input, ref output, Empty.Default, 0);
+
+            if (status == Status.PENDING)
             {
```
Comment (shargon) [inline src/FasterDBStore/Store.cs]: We don't need to check the Upsert status?

Comment (bettybao1209) [reply src/FasterDBStore/Store.cs]: we should check the status of upsert and delete, since they are the operation type that can be pending.

Comment (shargon) [reply src/FasterDBStore/Store.cs]: Before read it? two checks or with one it's enough?

Comment (bettybao1209) [reply src/FasterDBStore/Store.cs]: I think checking the read status after the upsert/delete operation should be ok

**Comment by shargon** [comment]
We can merge it and improve it in the other PR

**Pull request merged**